docs/theory/diffusion/diffusion.rst

```diff
@@ -0,0 +1,667 @@
+In addition to that, MSDs for different lags are highly correlated.
+One should not fit these values as though they were independent data points unless care is taken that it is valid to do so (more on this below).
+
+Ordinary Least Squares
+^^^^^^^^^^^^^^^^^^^^^^
+
+Real measurements are affected by noise, which leads to localization error.
+In addition to this, depending on the method used to detect the position of the particle, there may also be motion blur.
+These two sources of error manifest themselves as an offset in the MSD curve given by:
+
+.. math::
+
+    offset = 2 \sigma^2 - 4 R D \Delta t
+
+Here :math:`\sigma` is the static localization accuracy, :math:`R` is a motion blur constant and :math:`\Delta t` represents the time step.
+With pure diffusive motion (a complete absence of drift) in an isotropic medium, 1-dimensional MSDs can be fitted by the following relation:
+
+.. math::
+
+    \rho[n] = 2 D n \Delta t + offset
+
+Here :math:`D` is the diffusion constant in :math:`um^2/s`, :math:`\Delta t` is the time step, :math:`n` is the step index and the offset is determined by the localization accuracy.
+
+While it may be tempting to use a large number of lags in the diffusion estimation procedure, this actually produces poor estimates of the diffusion constant :cite:`qian1991single,michalet2010mean,michalet2012optimal`.
+There exists an optimal number of lags to fit such that the estimation error is minimal.
+This optimal number of lags depends on the ratio between the diffusion constant and the dynamic localization accuracy:
+
+.. math::
+
+    \epsilon_{localization} = \frac{offset}{slope} = \frac{2 \sigma^2 - 4 R D \Delta t}{2 D \Delta t} = \frac{\sigma^2}{D \Delta t} - 2 R
+
+When localization is infinitely accurate, the optimal number of points is two :cite:`michalet2010mean`.
+At the optimal number of lags, it doesn't matter whether we use a weighted or unweighted least squares algorithm to fit the curve :cite:`michalet2010mean`, and therefore we opt for the latter, analogously to :cite:`michalet2012optimal`.
+With Pylake, you can obtain an estimated diffusion constant by invoking::
+
+    >>> tracks[0].estimate_diffusion(method="ols")
+    DiffusionEstimate(value=7.804440367653842, std_err=2.527045387449447, num_lags=2, num_points=80, method='ols', unit='um^2 / s')
+
+Note that Pylake gives you both an estimate for the diffusion constant, as well as its expected uncertainty and the number of lags used in the computation.
+The uncertainty estimate in this case is based on Equation A1b in :cite:`bullerjahn2020optimal`.
+
+Generalized Least Squares
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Another option is to use generalized least squares (GLS).
+Based on the definition of the MSD estimator, it is possible to derive the structure of the covariance matrix :cite:`bullerjahn2020optimal`.
+This covariance matrix can then be used in the estimation procedure to determine the diffusion constant.
```

Comment:
I am missing the reasoning as to why this method is useful. What about:

As mentioned above, subsequent data points in an MSD curve are highly correlated. One can account for these correlations by computing the covariance matrix of the MSD values :cite:`bullerjahn2020optimal`. This covariance matrix can then be used in the estimation procedure to determine the diffusion constant.

Reply:
Yeah, that sounds a lot better.

In practice, the performance of that method was somewhat disappointing to me. It seems that in most cases, it doesn't perform better than OLS if you just stick to the optimal number of points suggested by Michalet. But it takes a lot longer to compute.

If you include more points than considered optimal, then GLS will perform better than doing OLS with too many points, but I'm not entirely sure how often that is useful. Especially considering the additional constraints GLS includes (tracks may not have gaps).

I deliberately kept this section somewhat short to emphasize the other two methods.

Reply:
I see, indeed a good idea to keep it short. And I see you also advice OLS in the concluding remarks. Looks good like this.